src/carma-platform/parse_ros_bags.py

```diff
@@ -74,6 +145,67 @@ def get_detected_objects(ros_bag_file, output_file, time_offset):
     )
     print(f"Calculated sim time offset: '{time_offset}'")
 
+def get_detected_objects_with_sim_received_time(ros_bag_file, output_file):
+
+    class ObjectType(Enum):
+        UNKNOWN = 0
+        SMALL_VEHICLE = 1
+        LARGE_VEHICLE = 2
+        MOTORCYCLE = 3
+        PEDESTRIAN = 4
+        def __str__(self):
+            return self.name
+
+    detected_obj_topic_name = "/environment/fused_external_objects"
+
+    object_msgs_and_system_times = get_object_msgs_with_its_system_times(ros_bag_file, detected_obj_topic_name)
+
+    output_file.parent.mkdir(exist_ok=True)
+
+    if output_file.exists():
+        print(f"Output file {output_file} already exists. Overwriting file.")
+
+    sim_times = []
+    with rosbag.Bag(ros_bag_file, "r") as bag:
+        for _, msg, t in bag.read_messages(topics=["/sim_clock"]):
+            sim_times.append((t, msg.clock)) # Assuming 'clock' attribute holds simulation time in seconds
+
+    with open(output_file, "w", newline="") as file:
+        writer = csv.writer(file)
+        writer.writerow(
+            ["Received Simulation Time (ms)", "System Time (s)", "Object Id", "Object Type"]
```

Comment:
We should specify which simulation time this is (CDASim vs MOSAIC).

Reply:
There is no difference between the two no? Oh you mean CARLA vs CDASim time?